### ISSUE TITLE
_utf8_chunk and _tail fails on Python 3

### ISSUE THREAD

**Issue by alexandrevicenzi** [release-engineering/kobo#90]
`_utf8_chunk` and `_tail fails` raises an error:

     TypeError: memoryview: a bytes-like object is required, not 'str'

These functions also have different returns on Python 2 and 3. For py2 it returns `str` for py3 it returns `bytes`, this must be due to unicode errors.

**Comment by alexandrevicenzi** [comment]
Well, the thing is that there's a try except and if for some reason decode fails to encode into utf-8 it just returns the data that was passed.

If decode fails other functions that use `_utf8_chunk` in py3 will fail because the type of data is not compatible.

**Comment by yontalcar** [comment]
Can you explain where the problem is? To me it just seems that the test code is broken for Py3.

Also I noticed the following snippet:
```
self.assertEquals(content, content)
self.assertEquals(task_log.cache[filename], content)
self.assertEquals(task_log.changed[filename], False)
```

Both are fixed in my work branch in commit 6dda5e1

**Comment by lzaoral** [comment]
@rohanpm This issue was already fixed in ee9f506. Can you close it, please?
